File: src/main/java/com/codingforcookies/betterrecords/client/models/ModelLGSpeaker.java
package com.codingforcookies.betterrecords.src.client.models;

import net.minecraft.client.model.ModelBase;
import net.minecraft.client.model.ModelRenderer;
import net.minecraft.entity.Entity;

public class ModelLGSpeaker extends ModelBase {
	ModelRenderer wool;
	ModelRenderer base;

	public ModelLGSpeaker() {
		textureWidth = 128;
		textureHeight = 64;

		wool = new ModelRenderer(this, 48, 0);
		wool.addBox(-5F, -9F, -6.5F, 10, 22, 1);
		wool.setRotationPoint(0F, 10F, 0F);
		wool.setTextureSize(128, 64);
		wool.mirror = true;
		setRotation(wool, 0F, 0F, 0F);
		
		base = new ModelRenderer(this, 0, 0);
		base.addBox(-6F, 0F, -6F, 12, 24, 12);
		base.setRotationPoint(0F, 0F, 0F);
		base.setTextureSize(128, 64);
		base.mirror = true;
		setRotation(base, 0F, 0F, 0F);
	}

	public void render(Entity entity, float f, float f1, float f2, float f3, float f4, float f5) {
		super.render(entity, f, f1, f2, f3, f4, f5);
		setRotationAngles(f, f1, f2, f3, f4, f5, entity);
		wool.render(f5);
		base.render(f5);
	}

	private void setRotation(ModelRenderer model, float x, float y, float z) {
		model.rotateAngleX = x;
		model.rotateAngleY = y;
		model.rotateAngleZ = z;
	}
}

File: src/main/kotlin/com/codingforcookies/betterrecords/client/model/ModelLGSpeaker.kt
package com.codingforcookies.betterrecords.client.model

import net.minecraft.client.model.ModelBase
import net.minecraft.client.model.ModelRenderer
import net.minecraft.entity.Entity
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

@SideOnly(Side.CLIENT)
class ModelLGSpeaker : ModelBase() {
    var wool: ModelRenderer
    var base: ModelRenderer

    init {
        textureWidth = 128
        textureHeight = 64

        this.wool = ModelRenderer(this, 176, 64).apply {
            setRotationPoint(0.0F, 0.0F, 0.0F)
            addBox(-5.0F, 1.0F, -6.5F, 10, 22, 1)
            mirror = true
        }
        this.base = ModelRenderer(this, 128, 64).apply {
            setRotationPoint(0.0F, 0.0F, 0.0F)
            addBox(-6.0F, 0.0F, -6.0F, 12, 24, 12)
        }
    }

    override fun render(entity: Entity?, limbSwing: Float, limbSwingAmount: Float, ageInTick: Float, rotationYaw: Float, rotationPitch: Float, scale: Float) {
        this.wool.render(scale)
        this.base.render(scale)
    }

    fun setRotationAngles(modelRenderer: ModelRenderer, x: Float, y: Float, z: Float) {
        modelRenderer.rotateAngleX = x
        modelRenderer.rotateAngleY = y
        modelRenderer.rotateAngleZ = z
    }
}
